nucypher-core-wasm/tests/wasm.rs

```diff
@@ -705,54 +721,68 @@ fn threshold_decryption_request() {
 
     assert_eq!(encrypted_request_from_bytes, encrypted_request);
     assert_eq!(encrypted_request_from_bytes.ritual_id(), ritual_id);
-
-    let e2e_request = encrypted_request_from_bytes
-        .decrypt(&request_secret)
-        .unwrap();
     assert_eq!(
-        response_encrypting_key.to_compressed_bytes(),
-        e2e_request.response_encrypting_key().to_compressed_bytes()
+        encrypted_request_from_bytes.requester_public_key(),
+        requester_public_key
     );
-    assert_eq!(request, e2e_request.decryption_request());
 
-    // wrong secret key used
-    assert!(encrypted_request_from_bytes
-        .decrypt(&response_secret)
-        .is_err());
+    // service decrypts request
+    let service_shared_secret =
+        service_secret.diffie_hellman(&encrypted_request_from_bytes.requester_public_key());
```

Comment:
Ok, makes sense.... but does it allow somehow to read the bytes?

Reply:
I wasn't sure if to allow the `RequestSharedSecret` object to be serializable or not, from a security perspective, since it is a derived secret. 

Figured we don't really want it stored, transmitted anywhere, so at the moment there is no public `to_bytes()` function. That being said the underlying `x25519-dalek::SharedSecret` object does have a `to_bytes()` function, which I can expose if needed.

wdyt?

Reply:
Functionally, the test does ensure that the one shared secret (derived by the requester) is used to encrypt, and that the other shared secret (derived by the service) can decrypt it.

Reply:
✔️  - added to test on the rust side  (`dkg.rs`)- non-bindings.